lib/go/thrift/simple_server.go

```diff
@@ -229,12 +247,29 @@ func (p *TSimpleServer) Serve() error {
 func (p *TSimpleServer) Stop() error {
 	p.mu.Lock()
 	defer p.mu.Unlock()
+
 	if atomic.LoadInt32(&p.closed) != 0 {
 		return nil
 	}
 	atomic.StoreInt32(&p.closed, 1)
 	p.serverTransport.Interrupt()
-	p.wg.Wait()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	go func() {
+		defer cancel()
+		p.wg.Wait()
+	}()
+
+	if ServerCloseTimeout > 0 {
+		select {
+		case <-time.After(ServerCloseTimeout):
```

Comment:
Say `ServerCloseTimeout` is 1min and all clients closed after 5s, so we go with the other case, this timer will still occupy the resources for the whole 1min and only gets recycled after the timer has fired, even though it's no longer needed after 5s. This is why we should defer `timer.Stop()` so we can release the resources sooner.